### PR TITLE
upgrade: deno_doc, deno_lint, dprint, swc

### PR DESCRIPTION
**Pull request by bartlomieju**
Fixes https://github.com/denoland/deno/issues/8180
Fixes https://github.com/denoland/deno/issues/8188
Fixes https://github.com/denoland/deno/issues/8189
Fixes https://github.com/denoland/deno/issues/8148

### CODE REVIEW

File: cli/tests/bundle/fixture11.out [hunk 1]
```diff
@@ -6,9 +6,9 @@ var O;
     O1[O1["A"] = 0] = "A";
     O1[O1["B"] = 1] = "B";
     O1[O1["C"] = 2] = "C";
-})((void 0) || (O = {
+})(O || (O = {
 }));
-const O1 = void 0;
+const O1 = O;
```
Comment (bartlomieju) [inline cli/tests/bundle/fixture11.out:11]: @kitsonk please take a look - the result of this test changed after the upgrade, I'm not sure if that's ok

Comment (bartlomieju) [reply cli/tests/bundle/fixture11.out:11]: Also CC @kdy1

Comment (kdy1) [reply cli/tests/bundle/fixture11.out:11]: I think it's ok. Actually previous output is a bug.

Previously exported `O` was undefined, but it should be `O1`, which is enum.

Comment (kitsonk) [reply cli/tests/bundle/fixture11.out:11]: Yeah, it was a big but we didn't notice it.

**Review by kitsonk — APPROVED** [review]
LGTM

**Pull request merged**